rayon-core/src/join/mod.rs

```diff
@@ -151,23 +151,21 @@ where
         // pushed on top of it in the stack, and we will have to pop
         // those off to get to it.
         while !job_b.latch.probe() {
-            if let Some(job) = worker_thread.take_local_job() {
-                if job_b_id == job.id() {
-                    // Found it! Let's run it.
-                    //
-                    // Note that this could panic, but it's ok if we unwind here.
-                    let result_b = job_b.run_inline(injected);
-                    return (result_a, result_b);
-                } else {
-                    worker_thread.execute(job);
-                }
-            } else {
+            let Some(job) = worker_thread.take_local_job() else {
```

Comment:
Any particular reason to switch to let-else here? At least to me, it isn't immediately obvious why the new way of writing this is clearer.

Reply:
No _big_ reason, but it at least reduces indentation. I think it's a little more clear which path is going to continue the loop too.

Reply:
Fair enough. Personally, I would have gone for a three-arm match with a guard for the job ID. But as you said, this looks like a primarily stylistic question.

The rest of changes look good to me.